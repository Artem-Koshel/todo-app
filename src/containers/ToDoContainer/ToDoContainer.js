import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import Input from '../../components/Input/Input';
import List from '../../components/List/List';
import styles from './ToDoContainer.module.scss';
import { addItem } from '../../redux/slices/todoListActions';
import ToDoItem from '../ToDoItem/ToDoItem';
import ControlBar from '../ControlBar/ControlBar';

export default function ToDoContainer() {
  const items = useSelector((state) => state.todoList);
  const selectedFilter = useSelector((state) => state.selectedFilter);

  const filteredItems =
    selectedFilter === 'All'
      ? items
      : items.filter((item) => item.done === (selectedFilter === 'Completed'));

  const dispatch = useDispatch();

  const addTodoListItem = (text) => {
    dispatch(
      addItem({
        done: false,
        text,
      })
    );
  };

  return (
    <div className={styles.todoContainer}>
      <h1>TODO</h1>
      <Input onEnterPress={addTodoListItem} />
      <List items={filteredItems} Item={ToDoItem} ControlBar={ControlBar} />
    </div>
  );
}
