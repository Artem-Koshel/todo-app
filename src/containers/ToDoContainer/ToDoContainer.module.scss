.todoContainer {
    position: absolute;
    top: 50px;
    width: 60%;
    display: flex;
    flex-direction: column;
    left: 20%;
    font-size: large;

    h1 {
        color: white;
        font-family: Arial, Helvetica, sans-serif;
        font-weight: bold;
        font-style: normal;
        letter-spacing: .5rem;
        text-align: left;
    }
}

