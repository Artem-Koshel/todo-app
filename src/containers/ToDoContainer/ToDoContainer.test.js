import React from 'react';
import renderWithProviders from '../../utils/test-utils';
import ToDoContainer from './ToDoContainer';

const mockInput = jest.fn();
jest.mock('../../components/Input/Input', () => (props) => {
  mockInput(props);
  return <div data-testid="input" />;
});

const mockList = jest.fn();
jest.mock('../../components/List/List', () => (props) => {
  mockList(props);
  return <div data-testid="list" />;
});

jest.mock('../ToDoItem/ToDoItem', () => 'mockTodoItem');
jest.mock('../ControlBar/ControlBar', () => 'mockControlBar');

describe('ToDoContainer component', () => {
  it('renders correctly', () => {
    const items = [1, 2, 3];
    const { container } = renderWithProviders(<ToDoContainer />, {
      preloadedState: { todoList: items },
    });
    expect(mockList).toBeCalledWith(
      expect.objectContaining({
        Item: 'mockTodoItem',
        ControlBar: 'mockControlBar',
        items,
      })
    );
    expect(mockInput).toBeCalledWith(
      expect.objectContaining({
        onEnterPress: expect.any(Function),
      })
    );
    expect(container).toMatchSnapshot();
  });
});
