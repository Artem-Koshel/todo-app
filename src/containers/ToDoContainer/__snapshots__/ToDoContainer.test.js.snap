// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoContainer component renders correctly 1`] = `
<div>
  <div
    class="todoContainer"
  >
    <h1>
      TODO
    </h1>
    <div
      data-testid="input"
    />
    <div
      data-testid="list"
    />
  </div>
</div>
`;
